src/components/cl/hier/allgatherv/allgatherv.c

```diff
@@ -120,6 +120,30 @@ static inline ucc_status_t is_host_ordered(ucc_cl_hier_team_t *cl_team, int *ord
     return UCC_OK;
 }
 
+/* Check if the displacements are contig, whether or not the user passed
+   UCC_COLL_ARGS_FLAG_CONTIG_DST_BUFFER in coll args */
+static inline ucc_status_t is_disp_contig(ucc_cl_hier_team_t *cl_team, ucc_base_coll_args_t *coll_args, int *contig)
```

Comment:
i think it's better to move this function to coll utils so that it can be reused by other TLs. For instance it can be used to enable knomial allgatherv.